### PR TITLE
test: cover checkpoint, model api due to invariant changes [DET-4980]

### PR DESCRIPTION
**Pull request by stoksc**
## Description
Previously checkpoints and validations were always aligned, now they are not. This just makes sure the model DB queries are OK with that.

## Test Plan
- [x] is a test

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ioga — APPROVED** [review]

File: master/static/srv/get_model_version.sql [hunk 1]
```diff
@@ -27,6 +27,7 @@ c AS (
     COALESCE(c.framework, '') as framework,
     COALESCE(c.format, '') as format,
     COALESCE(c.determined_version, '') as determined_version,
+    (v.metrics->'validation_metrics'->>(e.config->'searcher'->>'metric'))::float8 AS searcher_metric,
```
Comment (stoksc) [inline master/static/srv/get_model_version.sql:30]: tests found these were missing, @ioga of some interest maybe

Comment (ioga) [reply master/static/srv/get_model_version.sql:30]: cc @mapmeld

Comment (stoksc) [reply master/static/srv/get_model_version.sql:30]: these and other recent issues feel like strong justification for using an orm, or at least something that autogens types from sql queries. (sqlc)

Comment (ioga) [reply master/static/srv/get_model_version.sql:30]: I'd be amazed if sqlc is capable of correctly processing these

Comment (stoksc) [reply master/static/srv/get_model_version.sql:30]: Yeah, same. I took a look at their issues and it looks like it's far from perfect. But still is cool.

**Comment by stoksc** [comment]
just gonna land this around harness flakes

**Pull request merged**